compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -525,7 +524,7 @@ class PlainPrinter(_ctx: Context) extends Printer {
     case '"' => "\\\""
     case '\'' => "\\\'"
     case '\\' => "\\\\"
-    case _ => if (ch.isControl) "\\0" + toOctalString(ch) else String.valueOf(ch)
+    case _ => if (ch.isControl) "\\u%04x".format(ch.toInt) else String.valueOf(ch)
```

Comment:
f interpolator, cf https://github.com/lampepfl/dotty/pull/13136/files#diff-265361a32415e25bed50400e4bb357fb31b82ac9f3640dd7a6c9a3599b7701c9R727

Reply:
Ok, changed that both in `PlainPrinter` and `SourceCode`.